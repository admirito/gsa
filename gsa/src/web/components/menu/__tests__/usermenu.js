/* Copyright (C) 2018-2020 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import React from 'react';

import date from 'gmp/models/date';
import {longDate} from 'gmp/locale/date';

import {setSessionTimeout, setUsername} from 'web/store/usersettings/actions';

import {fireEvent, rendererWith} from 'web/utils/testing';

import UserMenu from '../usermenu';

describe('UserMenu component tests', () => {
  test('should render UserMenu', () => {
    const {render} = rendererWith({gmp: {}, router: true, store: true});

    const {element} = render(<UserMenu />);

    expect(element).toMatchSnapshot();
  });

  test('should render username and sessionTimeout', () => {
    const {render, store} = rendererWith({gmp: {}, router: true, store: true});
    const timeout = date('2018-10-10');

    store.dispatch(setSessionTimeout(timeout));
    store.dispatch(setUsername('foo'));

    const {element} = render(<UserMenu />);

    expect(element).toHaveTextContent(longDate(timeout));
    expect(element).toHaveTextContent('foo');
  });

  test('should route to usersettings on click', () => {
    const {render, history} = rendererWith({
      gmp: {},
      store: true,
      router: true,
    });

    const {getByTestId} = render(<UserMenu />);
    const userSettingsElement = getByTestId('usermenu-settings');

    fireEvent.click(userSettingsElement);

    expect(history.location.pathname).toMatch('usersettings');
  });

  test('should logout user on click', () => {
    const doLogout = jest.fn().mockResolvedValue();
    const gmp = {
      doLogout,
    };
    const {render} = rendererWith({gmp, store: true, router: true});

    const {getByTestId} = render(<UserMenu />);
    const userSettingsElement = getByTestId('usermenu-logout');

    fireEvent.click(userSettingsElement);

    expect(gmp.doLogout).toHaveBeenCalled();
  });
});

// vim: set ts=2 sw=2 tw=80:
