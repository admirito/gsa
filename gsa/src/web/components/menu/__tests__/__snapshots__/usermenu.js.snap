// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu component tests should render UserMenu 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c8 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c8 > * {
  margin-left: 5px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c2 * {
  height: inherit;
  width: inherit;
}

.c9 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c9 * {
  height: inherit;
  width: inherit;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  position: relative;
  display: none;
  -webkit-animation: iKjwYD 0.1s ease-in;
  animation: iKjwYD 0.1s ease-in;
}

.c0:hover .c4 {
  display: block;
}

.c5 {
  position: absolute;
  margin: 0;
  padding: 0;
  right: 0;
  top: 0;
  z-index: 600;
  list-style: none;
  font-size: 10px;
  width: 255px;
}

.c6 {
  height: 30px;
  width: 255px;
  border-left: 1px solid #787878;
  border-right: 1px solid #787878;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding-left: 12px;
}

.c6:hover {
  background: #66c430;
  color: #fff;
  cursor: pointer;
}

.c6:first-child {
  border-top: 1px solid #787878;
  cursor: default;
  background-color: #eeeeee;
}

.c6:first-child:hover {
  color: #000;
}

.c6:nth-child(2) {
  cursor: default;
  background-color: #eeeeee;
}

.c6:nth-child(2):hover {
  color: #000;
}

.c6:last-child {
  border-top: 1px solid #787878;
  border-bottom: 1px solid #787878;
}

.c3 {
  margin-right: 10px;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 100%;
  height: 100%;
}

.c10:link,
.c10:hover,
.c10:active,
.c10:visited,
.c10:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<span
  class="c0 c1"
  data-testid="usermenu"
>
  <span
    class="c2 c3"
    data-testid="svg-icon"
  >
    <svg>
      user.svg
    </svg>
  </span>
  <div
    class="c4"
  >
    <ul
      class="c5"
    >
      <li
        class="c6"
        title="Logged in as: "
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            margin="5px"
          >
            <span
              class="c9"
              data-testid="svg-icon"
            >
              <svg>
                user.svg
              </svg>
            </span>
            <span />
          </div>
        </div>
      </li>
      <li
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            margin="5px"
          >
            <span
              class="c9"
              data-testid="svg-icon"
            >
              <svg>
                schedule.svg
              </svg>
            </span>
            <span>
              Session timeout: 
            </span>
          </div>
        </div>
      </li>
      <li
        class="c6"
      >
        <a
          class="c10"
          data-testid="usermenu-settings"
          href="/usersettings"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
              margin="5px"
            >
              <span
                class="c9"
                data-testid="svg-icon"
              >
                <svg>
                  my_setting.svg
                </svg>
              </span>
              <span>
                My Settings
              </span>
            </div>
          </div>
        </a>
      </li>
      <li
        class="c6"
        data-testid="usermenu-logout"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            margin="5px"
          >
            <span
              class="c9"
              data-testid="svg-icon"
            >
              <svg>
                logout.svg
              </svg>
            </span>
            <span>
              Log Out
            </span>
          </div>
        </div>
      </li>
    </ul>
  </div>
</span>
`;
